pkg/cluster/delete.go

```diff
@@ -438,10 +424,15 @@ func (m *manager) deleteFederatedCredentials(ctx context.Context) error {
 				federatedIdentityCredentialResourceName,
 				&armmsi.FederatedIdentityCredentialsClientDeleteOptions{},
 			)
-
-			// As long as the customer can clean up resources on their own, we don't block deletion on failure to clean up federated credentials
 			if err != nil {
-				m.log.Errorf("failed to delete federated identity credentials for %s: %v", identity.ResourceID, err)
+				cloudErr := err.(*api.CloudError)
+
+				if cloudErr.StatusCode != http.StatusNotFound {
```

Comment:
Should this be `==` instead of `!=`?

Reply:
That's right, not sure where my brain was at that time :)